### PR TITLE
Androidx migration

### PR DESCRIPTION
**Pull request by pueffl**
Migrate to AndroidX.

Migration to AndroidX through AS migration tool and a few necessary fixes. This pull request is separated in 3 commits:

-) Migration to AndroidX through the AS migration tool.
-) Minor fixes after the migration.
-) Merge master

Note: currently some libs are alpha versions.

force "androidx.lifecycle:lifecycle-livedata-core:2.1.0-alpha01"
force "androidx.lifecycle:lifecycle-runtime:2.1.0-alpha01"
force "androidx.lifecycle:lifecycle-process:2.1.0-alpha01"
force "androidx.lifecycle:lifecycle-extensions:2.1.0-alpha01"
force "androidx.core:core:1.1.0-alpha03"

 It may be possible to downgrade this but it would be necessary to force a lot more versions and it comes from the migration tool. As the libs come from Google, I think we will get rid of them pretty soon.

### CODE REVIEW

**Comment by pueffl** [comment]
> Using an alpha variant can introduce unknown bugs. All of these have stable equivalents that should be used instead.

Thank you for your comment. I would usually agree to you but as I mentioned, I left this to the alpha versions because most of the migration was done by Googles tools and the versions where not introduced by me (I suppose that there is a reason for doing so at Google, so I didn't try tro downgrade it, also because I would have to specify/force more versions explicitely which may introduce even be more version incompatibilities). And as this comes from Google, we should see a stable version soon.

This migration is also just the first of 3 big migrations (AndroidX, Kotlin (including coroutines) and Koin) which may introduce some problems. As far as I can see, everything works fine but it would be good to do some more tests and to do them after all three migrations (to save time). From my experience such migrations work usually quite well, maybe there are some problems on some phones/versions but I wouldn't expect serious problems.

**Comment by dri94** [comment]
Google just updates to the latest version it can find, which when using maven also includes alpha builds. Thats why it auto does that. And yes Koin does currently cause issues with stable version (I am actually impatiently waiting for your Kotlin and AndroidX PRs to make it in so i can do some updates to this codebase). However even if only a small number of devices are affected, it would then make the release unstable pending google getting the bugs fixed. I would say merging this into master could be dangerous, but creating a beta branch and possibly a beta release in the store would be a better route.

File: app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsFragment.kt [hunk 1]
```diff
@@ -332,7 +332,7 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), ConfirmationDialogFr
                 getString(R.string.cancel_action)
         )
         dialogFragment.setTargetFragment(this, REQUEST_DELETE_ACCOUNT)
-        dialogFragment.show(fragmentManager, TAG_DELETE_ACCOUNT_CONFIRMATION)
+        dialogFragment.show(fragmentManager!!, TAG_DELETE_ACCOUNT_CONFIRMATION)
```
Comment (dri94) [inline app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsFragment.kt]: using "!!" defeats the purpose of kotlin's null safety

File: build.gradle [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+                'compileSdk': 28,
                 'minSdk': 19,
                 'buildTools': '28.0.3'
         ]
 
         versions = [
                 'kotlin': '1.2.71',
-                'supportLibrary': '27.1.1',
-                'preferencesFix': '27.1.1.1',
+                'supportLibrary': '28.0.0',
```
Comment (dri94) [inline build.gradle]: Is this still necessary, all support library references should be removed. We should mix support library with androidX

**Comment by cketti** [comment]
I think this migration should be done in two steps. First upgrading to support library version 28. Then switching to androidx. That should make the pull requests easier to review.

Using alpha or beta versions of libraries is not something we want to do. Libraries from Google are not special in this regard.

Please rebase your branch on top of K-9 Mail's master branch instead of merging master into your branch.

File: app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsFragment.kt (hunk 1 above)
Comment (pueffl) [reply app/ui/src/main/java/com/fsck/k9/ui/settings/account/AccountSettingsFragment.kt]: Yes, this method or some part of the method code are no longer null safe and it wouldn't compile anymore.

**Comment by pueffl** [comment]
Hope, this is ok now.

Once again for your understanding:
- ed19c59  is the result of the automatic migration (you can review them if you don't trust the migration tool but I think the changes are ok)
- 9b38fdc are the changes I introduced to fix the build. They have to be reviewed but there are only a few.
- 2fa88c7 are just the changes for Travis.

File: build.gradle (hunk 2 above)
Comment (pueffl) [reply build.gradle]: I tried just to do the minimum necessary changes to keep the changes small. There may be some additional refactorings necessary after the migration.

**Comment by cketti** [comment]
> Hope, this is ok now.

Unfortunately no. The branch still contains unrelated commits and merge commits. Try using one of the tools that visualize the commit graph and I'm sure you see what I mean.
Also, alpha versions of androidx libraries are still used.

I went ahead and did the conversion myself. Closing this PR in favor of #3907.

**Pull request closed without merging**